clouds/aws/utilities/delete_aws_resources.py

```diff
@@ -0,0 +1,193 @@
+            # For attached policies, we need to detach them using ARN
+            attached_role_policies_dict = _iam_client.list_attached_role_policies(
+                RoleName=role_name, MaxItems=1000
+            )
+            for attached_policy in attached_role_policies_dict["AttachedPolicies"]:
+                attached_policy_name = attached_policy["PolicyName"]
+                LOGGER.info(
+                    f"Detach policy {attached_policy_name} for role {role_name}"
+                )
+                _iam_client.detach_role_policy(
+                    RoleName=role_name, PolicyArn=attached_policy["PolicyArn"]
+                )
+
+            # For detached policies, we need to delete them by name
+            detached_policy_names_dict = _iam_client.list_role_policies(
+                RoleName=role_name, MaxItems=1000
+            )
+            for detached_policy_name in detached_policy_names_dict["PolicyNames"]:
+                LOGGER.info(
+                    f"Delete policy {detached_policy_name} for role {role_name}"
+                )
+                _iam_client.delete_role_policy(
+                    RoleName=role_name, PolicyName=detached_policy_name
+                )
+
+            LOGGER.info(f"Delete role {role_name}")
+            _iam_client.delete_role(RoleName=role_name)
+
+    return any(
+        [
+            roles_dict["IsTruncated"],
+            attached_role_policies_dict.get("IsTruncated"),
+            detached_policy_names_dict.get("IsTruncated"),
+        ]
+    )
+
+
+def delete_buckets(region_name):
+    LOGGER.info("Executing delete_buckets")
+    _s3_client = s3_client(region_name=region_name)
+    buckets_dict = _s3_client.list_buckets(MaxItems=MAX_ITEMS)
+    for bucket in buckets_dict["Buckets"]:
+        delete_all_objects_from_s3_folder(
+            bucket_name=bucket["Name"], boto_client=_s3_client
+        )
+        delete_bucket(bucket_name=bucket["Name"], boto_client=_s3_client)
+
+    return buckets_dict.get("IsTruncated")
+
+
+@click.command()
+@click.option(
+    "--aws-regions",
+    help="""
+        \b
+        Comma-separated string of AWS regions to delete resources from.
+        If not provided will iterate over all regions.
+        """,
+    required=False,
+)
+@click.option(
+    "--all-aws-regions",
+    help="""
+        \b
+        Run on all AWS regions.
+        """,
+    is_flag=True,
+)
+def main(aws_regions, all_aws_regions):
+    if all_aws_regions:
+        _aws_regions = aws_region_names()
+    elif aws_regions:
+        _aws_regions = aws_regions.split(",")
+    else:
+        click.echo("Either pass --all-aws-regions or --aws-regions to run cleanup")
+        raise click.Abort()
+
+    verify_aws_credentials()
+
+    rerun_cleanup_list = clean_aws_resources(aws_regions=_aws_regions)
+    while rerun_cleanup_list:
+        rerun_cleanup_list = clean_aws_resources(aws_regions=rerun_cleanup_list)
+
+
+def clean_aws_resources(aws_regions):
```

Comment:
Please add docstring that explain what is done here.

Reply:
done